src/routes/signup/+page.server.ts

```diff
@@ -34,11 +35,18 @@ export const actions: Actions = {
       }
     } catch (error) {
       if (error instanceof ZodError) {
-        const errors = error.flatten().fieldErrors
+        return fail(422, {
+          fields,
+          errors: error.flatten().fieldErrors,
+        })
+      }
 
+      if (error instanceof PrismaClientKnownRequestError && error?.code === 'P2002') {
```

Comment:
another way would be to use the preprocess:

```ts
    .refine((data) => {
      await emailExists(data.email) // prisma query to the user table
    }, {
          message: "Email already exists.",
          path: ['email'], // path of error
        })
```

But that's maybe overkill and this is also nice, so NVM.